js/modules/k6/ws/ws.go

```diff
@@ -104,9 +107,20 @@ func (*WS) Connect(ctx context.Context, url string, args ...goja.Value) (*WSHTTP
 
 	// Leave header to nil by default so we can pass it directly to the Dialer
 	var header http.Header
+	var jar http.CookieJar
 
 	tags := state.CloneTags()
 
+	u, err := netURL.Parse(url)
+	if err != nil {
+		return nil, err
+	}
+
+	cookiesMap := make(map[string]http.Cookie)
+	for _, cookie := range state.CookieJar.Cookies(u) {
```

Comment:
Should we write the `cookiesMap` changes back to `state.CookieJar`? cc: @MStoykov 

And I think @MStoykov would like to reuse the cookie logic in the `httpext` package, e.g. [`SetRequestCookies`](https://github.com/loadimpact/k6/blob/4e637f6e57906584a5ae2f90da836802be4e03c9/lib/netext/httpext/request.go#L383), though we should probably agree on the details.

Reply:
Looking at the [websocket docs](https://godoc.org/github.com/gorilla/websocket) it looks like reusing it will be ... harder than I expected ...  read ... not possible unless we rewrite it which won't help at all either way.

Looking at the httpext k6 does write back *only* stuff that [were set in the responses](https://github.com/loadimpact/k6/blob/4e637f6e57906584a5ae2f90da836802be4e03c9/lib/netext/httpext/request.go#L283-L290)